src/main/java/com/cmsc495/ticketsystem/controller/AdminController.java

```diff
@@ -24,11 +29,50 @@ public class AdminController {
     @Autowired
     private TicketService ticketService;
 
+    @Autowired
+    private EmailService emailService;  // Email service to send email notifications
 
+    // Display all tickets on the admin page
     @GetMapping("/admin")
     public String showAdminPage(Model model) {
         List<TicketModel> ticketModelList = ticketService.findAllTickets();
         model.addAttribute("tickets", ticketModelList);
         return "admin";
     }
-}
+
+    // Method to close a ticket and send an email notification
+    @PostMapping("/close-ticket/{ticketId}")
```

Comment:
This functionality should be handled on the TicketController page.

Miles added in TicketController.updateTicketDetails which already handles closing a ticket if the remove checkbox is selected. We want to Autowire the EmailService into that as you have done already and call it if and when the ticket is deleted. You're on the right track but it's in the wrong area.

We can also handle sending a closure notification in that as well.